contrib/aws/aws-sdk-go-v2/aws/aws.go

```diff
@@ -0,0 +1,134 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package aws
+
+import (
+	"context"
+	"fmt"
+	"math"
+
+	awsmiddleware "github.com/aws/aws-sdk-go-v2/aws/middleware"
+	"github.com/aws/smithy-go/middleware"
+	smithyhttp "github.com/aws/smithy-go/transport/http"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+const (
+	tagAWSAgent     = "aws.agent"
+	tagAWSService   = "aws.service"
+	tagAWSOperation = "aws.operation"
+	tagAWSRegion    = "aws.region"
+	tagAWSRequestID = "aws.request_id"
+)
+
+// Middleware allows us to add middleware to the AWS SDK GO v2.
+// See https://aws.github.io/aws-sdk-go-v2/docs/middleware for more information.
+type Middleware struct {
```

Comment:
What's the reason to keep this struct? It is never used anywhere other than setting things up. This API can be one single exported function:
```go
func WrapConfig(cfg *aws.Config, opts ...Option) {
     cfg.APIOptions = append(cfg.APIOptions, ..., ...)
}
```
That's it?

Reply:
Yeah this is a fair point. That struct is just adding noise :)